src/Commands/RunCommand.cs

```diff
@@ -28,4 +32,108 @@ public override string GetCommandName()
 
     public string ScriptToRun { get; set; }
     public ScriptType Type { get; set; }
+    public int TimeoutMilliseconds { get; set; }
+
+    public async Task<(bool success, string output)> ExecuteWithTimeout()
```

Comment:
Good to add the property here, but, don't need this method... I think the `RunShellCommandAsync` method in `ProcessHelpers` already does the right thing. I think we just need to pass the timeout into it... via the way that `HandleRunCommand` used to be written (you should undo your change to it, or re-work it appropriately)/

You can run the run tests and make sure you're stuff works by building the dotnet app, and then running `ai test run --contains "name of test"` ... that'll produce a TRX output that will have error info if anything fails.